imperial_coldfront_plugin/views.py

```diff
@@ -51,3 +61,87 @@ def group_members_view(request: HttpRequest, user_pk: int) -> HttpResponse:
     group_members = GroupMembership.objects.filter(owner=user)
 
     return render(request, "group_members.html", {"group_members": group_members})
+
+
+@login_required
+def send_group_invite(request: HttpRequest) -> HttpResponse:
+    """Invite an individual to a group."""
+    if request.method == "POST":
+        form = GroupMembershipForm(request.POST)
+
+        if form.is_valid():
+            invitee_email = form.cleaned_data["invitee_email"]
+
+            # Create invitation URL.
+            signer = TimestampSigner()
+            token = signer.sign_object(
+                {
+                    "inviter_pk": request.user.pk,
+                    "invitee_email": invitee_email,
+                }
+            )
+            invite_url = request.build_absolute_uri(
+                reverse("imperial_coldfront_plugin:accept_group_invite", args=[token])
+            )
+
+            # Send invitation via email.
+            send_mail(
+                "You've been invited to a group",
+                f"Click the following link to accept the invitation:\n{invite_url}",
+                request.user.email,
+                [invitee_email],
+            )
+
+    else:
+        form = GroupMembershipForm()
+
+    return render(
+        request,
+        "imperial_coldfront_plugin/send_group_invite.html",
+        {"form": form},
+    )
+
+
+@login_required
+def accept_group_invite(request: HttpRequest, token: str) -> HttpResponse:
```

Comment:
One detail from the original issue that seems to be missing is this: "it will work even for users that have not accessed the system and so for whom there is no associated user account".

So this view cannot be `login_required`. However, we still need to check against the user... Unless we're relying on the fact people will sign in with SSO after clicking the link and thus create an account?

Reply:
It's a good point. The SSO email will need to be known upfront by the inviter if the invitee is required to eventually login with an Imperial account. @cc-a?

Reply:
@AdrianDAlessandro yes, we're relying on SSO here to create the account details and (with #51 merged) to get user uids.